java-checks/src/main/java/org/sonar/java/checks/CastArithmeticOperandCheck.java

```diff
@@ -103,6 +105,14 @@ public void visitBinaryExpression(BinaryExpressionTree tree) {
     }
   }
 
+  @Override
+  public void visitNewClass(NewClassTree tree) {
+    if (tree.is(Tree.Kind.NEW_CLASS)) {
```

Comment:
the `tree` is necessarily a `NewClassTree` here, so you already know it's going to be of the good kind. You can drop the `if`.